utility/src/main/java/org/oppia/android/util/networking/DebugNetworkConnectionUtil.kt

```diff
@@ -0,0 +1,45 @@
+package org.oppia.android.util.networking
+
+import org.oppia.android.util.networking.NetworkConnectionUtil.ConnectionStatus
+import javax.inject.Inject
+import javax.inject.Singleton
+
+/**
+ * [NetworkConnectionUtil] that gets and sets the current [ConnectionStatus] of the device in debug builds.
+ */
+@Singleton
+class DebugNetworkConnectionUtil @Inject constructor(
+  private val prodNetworkConnectionUtil: ProdNetworkConnectionUtil
+) : NetworkConnectionUtil {
+
+  private var forcedConnectionStatus: ConnectionStatus? = null
+
+  override fun getCurrentConnectionStatus(): ConnectionStatus {
+    val actualConnectionStatus = prodNetworkConnectionUtil.getCurrentConnectionStatus()
+    if (actualConnectionStatus == ConnectionStatus.NONE) {
+      forcedConnectionStatus = null
+    }
+    forcedConnectionStatus?.let {
+      return it
+    }
+    return actualConnectionStatus
+  }
+
+  /**
+   * Forces [connectionStatus] as the current connection status of the device and returns a
+   * [Boolean] indicating result.
```

Comment:
Indicating what result? We should be more clear in this documentation to help indicate to the caller when they can specifically expect true vs. false.

Reply:
KDoc updated. PTAL.